goagen/gen_client/generator.go

```diff
@@ -1087,7 +1087,7 @@ func (c *Client) {{ $funcName }}(ctx context.Context, path string{{ if .Params }
 			return nil, err
 		}
 	}
-{{ else if $att.Type.IsPrimitive }}	{
+{{ else if $att.Type.IsPrimitive }}	{{ $optional := not ($payload.IsRequired $name) }}	{{ if $optional }}if {{ printf "payload.%s" (goifyatt $att $name true) }} != nil{{ end }}{
```

Comment:
Did you try `{{ if not ($payload.IsRequired $name) }}`?

Reply:
I tried it, and got same result. I removed `$optional` from this template.